SourceX/storm/storm.cpp

```diff
@@ -21,10 +21,10 @@ bool directFileAccess = false;
 
 static std::string getIniPath()
 {
-	char path[DVL_MAX_PATH], file_path[DVL_MAX_PATH];
+	char path[DVL_MAX_PATH], file_path[DVL_MAX_PATH + 10];
 
 	GetPrefPath(path, DVL_MAX_PATH);
-	snprintf(file_path, DVL_MAX_PATH, "%sdiablo.ini", path);
+	snprintf(file_path, DVL_MAX_PATH + 10, "%sdiablo.ini", path);

```

Comment:
It's kind of odd to allow 10 bytes extra for this path, for other cases we simply truncate paths, this just happens to be in a place where the compiler is able to detect that long paths will be truncated. Any other way to avoid this (sans shortening `path` - 10).

Reply:
10 bytes is the length of "diablo.ini"

Reply:
Yeah, but it still is a bit odd to handle 260+10 when the path is for diablo.ini but only 260 when the path is for single_1.sv or diabdat.mpq.

Reply:
We can simply not use `snprintf` and `DVL_MAX_PATH` here, I'll push a commit.

Reply:
`snprintf` is used since it nicely truncates lines that are too long (instead of overflowing), the warning that the compiler is giving is the expected behavior, is there any way to suppress it instead?

Reply:
Other than disabling these warnings for the whole project, I don't know of one. Why're we truncating paths?

https://github.com/diasurgical/devilutionX/pull/298/commits/2da24a109af60b950b922a73a1007cf6d65d0899

Reply:
Looks good :+1: